packages/core/strapi/lib/services/entity-validator/blocks-validator.js

```diff
@@ -0,0 +1,143 @@
+    .of(inlineNodeValidator)
+    .min(1, 'Heading node children must have at least one Text or Link node')
+    .required(),
+});
+
+const quoteNodeValidator = yup.object().shape({
+  type: yup.string().equals(['quote']).required(),
+  children: yup
+    .array()
+    .of(inlineNodeValidator)
+    .min(1, 'Quote node children must have at least one Text or Link node')
+    .required(),
+});
+
+const codeBlockValidator = yup.object().shape({
+  type: yup.string().equals(['code']).required(),
+  syntax: yup.string().nullable(),
+  children: yup
+    .array()
+    .of(textNodeValidator)
+    .min(1, 'Quote node children must have at least one Text or Link node')
+    .required(),
+});
+
+const listItemNode = yup.object().shape({
+  type: yup.string().equals(['list-item']).required(),
+  children: yup.array().of(inlineNodeValidator).required(),
+});
+
+const listNodeValidator = yup.object().shape({
+  type: yup.string().equals(['list']).required(),
+  format: yup.string().equals(['ordered', 'unordered']).required(),
+  children: yup
+    .array()
+    .of(listItemNode)
+    .min(1, 'List node children must have at least one ListItem node')
+    .required(),
+});
+
+const imageNodeValidator = yup.object().shape({
+  type: yup.string().equals(['image']).required(),
+  image: yup.object().shape({
+    name: yup.string().required(),
+    alternativeText: yup.string().nullable(),
+    url: yup.string().required(),
+    caption: yup.string().nullable(),
+    width: yup.number().required(),
+    height: yup.number().required(),
+    formats: yup.object().required(),
+    hash: yup.string().required(),
+    ext: yup.string().required(),
+    mime: yup.string().required(),
+    size: yup.number().required(),
+    previewUrl: yup.string().nullable(),
+    provider: yup.string().required(),
+    provider_metadata: yup.mixed().nullable(),
+    createdAt: yup.string().required(),
+    updatedAt: yup.string().required(),
+  }),
+  children: yup.array().of(inlineNodeValidator).required(),
+});
+
+const blockNodeValidator = yup.lazy((value) => {
+  switch (value.type) {
+    case 'paragraph':
+      return paragraphNodeValidator;
+    case 'heading':
+      return headingNodeValidator;
+    case 'quote':
+      return quoteNodeValidator;
+    case 'list':
+      return listNodeValidator;
+    case 'image':
+      return imageNodeValidator;
+    case 'code':
+      return codeBlockValidator;
+    default:
+      return yup.mixed().test('invalid-type', 'Block node is of invalid type', () => {
+        return false;
+      });
+  }
+});
+
+// See: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const semverRegex =
+  /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/gm;
+const blocksValidator = yup.object({
+  version: yup.string().matches(semverRegex, 'Invalid version').required(),
```

Comment:
ℹ️ Feedback from Alex, don't use semver but something simpler (for example [docker compose versions](https://docs.docker.com/compose/compose-file/compose-versioning/)). We also don't necessarily have to add it now but instead could add it in a migration when we actually creating a new version of the schema. @remidej wdyt? Should we keep it now and update the version or remove the idea of version and deal with it later?

Reply:
I'm going to remove version for the alpha, we can discuss it more, and apply it in the beta or release the stable without and handle it in a migration